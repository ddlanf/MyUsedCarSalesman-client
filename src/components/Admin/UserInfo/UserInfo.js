import React, { Component } from 'react'
import UserService from '../../../services/users-api-service'

export default class UserInfo extends Component {

    blockOrUnblockUser(user_id){
        UserService.blockUser(user_id)
            .then(()=> this.props.updateUsers())
    }
    
    deleteUser(user_id){
        UserService.deleteUser(user_id)
        .then(() => this.props.updateUsers())
    }

    renderUserInfo(){
        const { users } = this.props
        const userTable = users.map(user => {
            return (
                <tr 
                    key={user.id}
                    >
                    <th>{user.id}</th>
                    <th>{user.user_name}</th>
                    <th>{user.password}</th>
                    <th>{user.first_name}</th>
                    <th>{user.last_name}</th>
                    <th>{user.email}</th>
                    <th>{user.date_created}</th>
                    <th>
                        <button 
                            onClick={() => this.blockOrUnblockUser(user.id)}
                            className="admin-block">
                            {(user.user_status === "Blocked" ? "Unblock" : "Block")}
                        </button>
                    </th>
                    <th>
                        <button 
                             onClick={() => this.deleteUser(user.id)}
                            className="admin-delete">
                            Delete
                        </button>
                    </th>
                </tr>
            )
        })

        return userTable
    }


    render() {
        return (
        <div className="users-info users">
            <h2>User Info</h2>
            <table className="admin-table users">
                <thead>
                    <tr className="about-users">
                        <th>ID</th>
                        <th>Username</th>
                        <th>Password</th>
                        <th>First name</th>
                        <th>Last name</th>
                        <th>Email</th>
                        <th>Date Created</th>
                        <th></th>
                        <th></th>
                    </tr>
                </thead>
                <tbody>
                    {this.renderUserInfo()}
                </tbody>
            </table>
        </div>
        )
    }
}

