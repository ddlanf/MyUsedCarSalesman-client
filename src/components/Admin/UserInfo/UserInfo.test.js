import React from 'react';
import ReactDOM from 'react-dom';
import UserInfo from './UserInfo';
import renderer from 'react-test-renderer';

describe('UserInfo component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserInfo
        users={[]} 
        updateUsers={function(){}}
        />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<UserInfo
            users={[]} 
            updateUsers={function(){}}
            />)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});