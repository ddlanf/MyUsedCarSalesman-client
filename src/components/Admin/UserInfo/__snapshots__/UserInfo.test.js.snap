// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders the UI as expected 1`] = `
<div
  className="users-info users"
>
  <h2>
    User Info
  </h2>
  <table
    className="admin-table users"
  >
    <thead>
      <tr
        className="about-users"
      >
        <th>
          ID
        </th>
        <th>
          Username
        </th>
        <th>
          Password
        </th>
        <th>
          First name
        </th>
        <th>
          Last name
        </th>
        <th>
          Email
        </th>
        <th>
          Date Created
        </th>
        <th />
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
